kernel/libs/atomic-integer-wrapper/src/lib.rs

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: MPL-2.0
```

Comment:
Add crate-level Rust doc and README so that people know how to use the proc macro without diving into the code.